src/assignments/orbit-report.rst

```diff
@@ -0,0 +1,414 @@
+   to control how the array is sorted. For more details see the `MDN description of sort using a compare
+   function <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Description>`_
+
+.. sourcecode:: typescript
+   :linenos:
+
+   sort(column: string): void {
+    // array.sort modifies the array, sorting the items based on the given compare function
+    this.satellites.sort(function(a: Satellite, b: Satellite): number {
+      if(a[column] < b[column]) {
+        return -1;
+      } else if (a[column] > b[column]) {
+        return 1;
+      }
+      return 0;
+     });
+   }
+
+
+8) Searching
+^^^^^^^^^^^^
+You are doing great! Only two more features to add. Next you will add a search feature.
+
+#. Add this HTML ``<div class="search-form"></div>`` in your ``app.component.html``.
+#. Add an ``<input>`` element insde the ``<div>``.
+#. Add a ``<button>`` element insde the ``<div>``.
+#. Add an Angular click handler to the ``<button>`` that when clicked calls ``search(searchTerm.value)``
+
+   * ``searchTerm`` being the local variable name for the ``<input>``
+
+#. Add an Angular keyup.enter handler to the ``<input>`` that when clicked calls ``search(searchTerm.value)``
+
+   * ``searchTerm`` being the local variable name for the ``<input>``
+
+#. Add a ``search`` method to the ``AppComponent`` class.
+
+   * The ``search`` method is provded below.
+
+.. sourcecode:: typescript
+   :linenos:
+
+   search(searchTerm: string): void {
+      let matchingSatellites: Satellite[] = [];
+      searchTerm = searchTerm.toLowerCase();
+      for(let i=0; i < this.sourceList.length; i++) {
+         let name = this.sourceList[i].name.toLowerCase();
+         if (name.indexOf(searchTerm) >= 0) {
+            matchingSatellites.push(this.sourceList[i]);
+         }
+      }
+      // assign this.displayList to be the the array of matching satellites
+      // this will cause Angular to re-make the table, but now only containg matches
+      this.displayList = matchingSatellites;
+   }
+
+Notice the usage of a new variable named ``displayList``. ``displayList`` should contain the ``Satellite`` objects
+that the user wants to see. That was previously ALL the satellites. Now the user can perform a search, which
+means they want to see the matching results. ``sourceList`` will contain ALL the ``Satellite`` objects. If you removed
+the ``Satellite`` objects from ``sourceList``, then the user could never see them again. Instead when the user
+performs a search, ``displayList`` will be populated with the all the matching ``Satellite`` objects in ``sourceList``.
+Matching is defined as ``satellite.name`` containing the search term.
+
+7. Add the ``displayList: Satellite[];`` property to the ``AppComponent`` class.
+
+   * Set ``displayList = []`` in the constructor.
+
+8. Pass in the ``displayList`` to the ``orbit-list-component``.
```

Comment:
Do you want them to add something new or overwrite what was already written?

Reply:
If adding it, it creates a second table. If overwriting it, then it creates a blank table because ``displayList`` is an empty array.

Reply:
``this.displayList`` is updated to be the value of ``matchingSatellites`` in the search function.  Step 8 instructs students to pass in ``displayList`` to the component via``<app-orbit-list [satellites]="displayList"></app-orbit-list>``. There should only be one table.

Reply:
Also there was a bug related to the table being empty before a search happened. I just added an instruction to the search section to fix that.

Reply:
Cool! The blank table when adding ``displayList`` was the origin of my confusion! Resolved!